django/cantusdb_project/main_app/tests/test_functions.py

```diff
@@ -1,20 +1,162 @@
+            return MockResponse(
+                status_code=200,
+                content=mock_cantusindex_data.mock_json_nextchants_a07763_content,
+                text=mock_cantusindex_data.mock_json_nextchants_a07763_text,
+                json=None,
+            )
+        else:  # imitating CI's behavior when a made-up Cantus ID is entered.
+            return MockResponse(
+                status_code=200,
+                content=bytes('["Cantus ID is not valid"]', encoding="utf-8-sig"),
+                text='["Cantus ID is not valid"]',
+                json=["Cantus ID is not valid"],
+            )
+    elif (
+        "https://cantusindex.uwaterloo.ca/json-cid/" in url
+        or "https://cantusindex.org/json-cid/" in url
+    ):
+        if url.endswith("/008349"):
+            return MockResponse(
+                status_code=200,
+                content=mock_cantusindex_data.mock_json_cid_008349_content,
+                text=mock_cantusindex_data.mock_json_cid_008349_text,
+                json=mock_cantusindex_data.mock_json_cid_008349_json,
+            )
+        elif url.endswith("/006928"):
+            return MockResponse(
+                status_code=200,
+                content=mock_cantusindex_data.mock_json_cid_006928_content,
+                text=mock_cantusindex_data.mock_json_cid_006928_text,
+                json=mock_cantusindex_data.mock_json_cid_006928_json,
+            )
+        elif url.endswith("/008411c"):
+            return MockResponse(
+                status_code=200,
+                content=mock_cantusindex_data.mock_json_cid_008411c_content,
+                text=mock_cantusindex_data.mock_json_cid_008411c_text,
+                json=mock_cantusindex_data.mock_json_cid_008411c_json,
+            )
+        elif url.endswith("/008390"):
+            return MockResponse(
+                status_code=200,
+                content=mock_cantusindex_data.mock_json_cid_008390_content,
+                text=mock_cantusindex_data.mock_json_cid_008390_text,
+                json=mock_cantusindex_data.mock_json_cid_008390_json,
+            )
+        elif url.endswith("/007713"):
+            return MockResponse(
+                status_code=200,
+                content=mock_cantusindex_data.mock_json_cid_007713_content,
+                text=mock_cantusindex_data.mock_json_cid_007713_text,
+                json=mock_cantusindex_data.mock_json_cid_007713_json,
+            )
+        elif url.endswith("/909030"):
+            return MockResponse(
+                status_code=200,
+                content=mock_cantusindex_data.mock_json_cid_909030_content,
+                text=mock_cantusindex_data.mock_json_cid_909030_text,
+                json=mock_cantusindex_data.mock_json_cid_909030_json,
+            )
+        else:  # imitating CI's behavior when a made-up Cantus ID is entered.
+            return MockResponse(
+                status_code=500,
+                content=None,
```

Comment:
`content` and `text` are not valid as `None` according to definition of `__init__` of `MockResponse`